src/Workspaces/Core/Portable/SolutionCrawler/SolutionCrawlerOptions.cs

```diff
@@ -9,26 +10,44 @@ internal static class SolutionCrawlerOptions
         /// <summary>
         /// Option to turn configure background analysis scope.
         /// </summary>
-        public static readonly Option<BackgroundAnalysisScope> BackgroundAnalysisScopeOption = new Option<BackgroundAnalysisScope>(
+        public static readonly PerLanguageOption<BackgroundAnalysisScope> BackgroundAnalysisScopeOption = new PerLanguageOption<BackgroundAnalysisScope>(
             nameof(SolutionCrawlerOptions), nameof(BackgroundAnalysisScopeOption), defaultValue: BackgroundAnalysisScope.Default,
-            storageLocations: new RoamingProfileStorageLocation($"TextEditor.BackgroundAnalysisScopeOption"));
+            storageLocations: new RoamingProfileStorageLocation($"TextEditor.%LANGUAGE%.Specific.BackgroundAnalysisScopeOption"));
 
         /// <summary>
         /// Enables forced <see cref="BackgroundAnalysisScope.Minimal"/> scope when low VM is detected to improve performance.
         /// </summary>
         public static bool LowMemoryForcedMinimalBackgroundAnalysis = false;
 
         public static BackgroundAnalysisScope GetBackgroundAnalysisScope(Project project)
-            => GetBackgroundAnalysisScope(project.Solution.Options);
+            => GetBackgroundAnalysisScope(project.Solution.Options, project.Language);
 
-        public static BackgroundAnalysisScope GetBackgroundAnalysisScope(OptionSet options)
+        public static BackgroundAnalysisScope GetBackgroundAnalysisScope(OptionSet options, string language)
         {
             if (LowMemoryForcedMinimalBackgroundAnalysis)
             {
                 return BackgroundAnalysisScope.Minimal;
             }
 
-            return options.GetOption(BackgroundAnalysisScopeOption);
+            switch (language)
+            {
+                case LanguageNames.CSharp:
+                case LanguageNames.VisualBasic:
+                    return options.GetOption(BackgroundAnalysisScopeOption, language);
```

Comment:
So are we ignoring previous settings of the open/closed diagnostics setting for C# and VB? We don't want to read those if the new one hasn't been set?

Reply:
There is no UI to set the older option for C# and VB though. Also, I am not sure how would we hook up option page setting based on 2 options. Current `BindToOption` approach simply connects the UI radio button to the new option. Having said that, we can potentially set this up somehow, but I am not sure if the effort is worth it.